### PR TITLE
🧪test lexical analyzer

### LINKED ISSUE #39: 🐞Envp Load
build on macos fails, probably because the extern envp pointer is defined differently.

### PR DESCRIPTION
**Pull request by lucienvb**
Closes #39 and #12

### CODE REVIEW

File: unit_test/src/lexer_cases.c [hunk 1]
```diff
@@ -6,95 +6,187 @@
...
 
-t_list	*lexer(const char *command_line, t_exitstatus *exit_status);
-
-void assert_lexer(char *command_line, char **expected)
+void assert_lexer_one(char *command_line, char **expected)
 {
     t_list *tokens;
-    t_list *head;
 
     tokens = lexer(command_line, &zero);
```
Comment (saladuit) [inline unit_test/src/lexer_cases.c:25]: Perhaps a good idea to add a assertion: cr_assert(tokens != NULL);

File: unit_test/src/lexer_cases.c [hunk 2]
```diff
@@ -6,95 +6,187 @@
...
+    ft_lstclear(&tokens, free);
 }
 
-Test(lexer, basic)
+Test(lexer, one_ls)
 {
     char *expected[] = {"ls", NULL};
-    assert_lexer("ls", expected);
+    assert_lexer_one("ls", expected);
 }
```
Comment (saladuit) [inline unit_test/src/lexer_cases.c:35]: I think it is a good idea to also test for an empty string as input: ""

Comment (saladuit) [reply unit_test/src/lexer_cases.c:35]: Usually one should consider the following for strings:

1. NULL
2. Empty
3. single char/words
5. two chars/words

**Review by saladuit — CHANGES_REQUESTED** [review]
Very well done for your first proper unit test set!

File: src/lexer/lexer.c [hunk 3]
```diff
@@ -68,10 +147,14 @@ t_list	*lexer(const char *command_line, t_exitstatus *exit_status)
 	t_list	*tokens;
 	char	*token;
```
Comment (saladuit) [inline src/lexer/lexer.c:149]: Probaly it is a good idea to add an assertion here: assert(command_line != NULL);

**Review by saladuit — APPROVED** [review]
